saleor/discount/utils.py

```diff
@@ -1036,6 +1074,21 @@ def get_current_products_for_rules(rules: "QuerySet[PromotionRule]"):
     return Product.objects.filter(Exists(variants.filter(product_id=OuterRef("id"))))
 
 
+def get_channels_for_rules(rules: "QuerySet[PromotionRule]"):
+    """Get currently assigned channels to promotions.
+
+    Collect all channels for are assigned to promotion rules.
+    """
+    PromotionRuleChannel = PromotionRule.channels.through
+    rule_channels = PromotionRuleChannel.objects.filter(
+        Exists(rules.filter(pk=OuterRef("promotionrule_id")))
+    )
+    channels = Channel.objects.filter(
+        Exists(rule_channels.filter(channel_id=OuterRef("id")))
+    )
```

Comment:
We can add `using` with replica, right?

Reply:
Looks like we can fully drop this function.